Qt6 support

Amalvir opened the pull request:
Hello!

This PR adds Qt6 support using [qgis/pyqgis4-checker](https://github.com/qgis/pyqgis4-checker) and get the plugin ready for QGIS 4.

Changes included:
- Updated Python code to ensure compatibility with QGIS 4
- Updated `metadata.txt`
- Updated `pyqtgraph` to the latest version
- Tested on QGIS 3.44.2 and QGIS 3.44.2 (Qt6 build), and compared with the previous behaviour : results are consistent

Everything looks good on my side, but feedback is welcome!

Amalvir committed:
Run pyqt5_to_pyqt6.py script

Amalvir committed:
Fix oversights

Amalvir committed:
Update metadata.txt

Amalvir committed:
Update pyqtgraph

nicogodet commented:
Increase minimum version (could you try at least on 3.40 and set it as minimum version ?)
Could you drop resources.py and use plain path instead ? (pyrcc6 no longer exists which increase the burdain to maintain a resource file even if PyQt6 still supports it.)

Beside these 2 comments, thank you for this PR!

Amalvir committed:
Remove unused imports, files and compilated resources

Amalvir committed:
Change qgisMinimumVersion to 3.34

Amalvir commented:
Changes included:
- Removed the old way of using resources by replacing in with path
- Removed unused imports which was delaying the loading of the plugin
- Tested it in 3.34 and everything is fine
- Set the minimum version to 3.34
- Deleted the unused ui file

vcloarec commented:
merged! I will try to update the plugin in the QGIS repo this evening.

vcloarec commented:
Thanks you both!

vcloarec commented:
published on the QGIS plugin repo, waiting for being approved.
